packages/core/src/routes/profile/utils/get-scoped-profile.ts

```diff
@@ -0,0 +1,69 @@
+import { UserScope } from '@logto/core-kit';
+import { type UserProfileResponse } from '@logto/schemas';
+import { conditional } from '@silverhand/essentials';
+
+import type Libraries from '../../../tenants/Libraries.js';
+import type Queries from '../../../tenants/Queries.js';
+import { transpileUserProfileResponse } from '../../../utils/user.js';
+
+/**
+ * Get the user profile, and filter the fields according to the scopes.
+ * The scopes and fields are defined in the core-kit, see packages/toolkit/core-kit/src/openid.ts
+ */
+export const getScopedProfile = async (
```

Comment:
Wondering how we can maintain an SSOT of the user scope to user info mapping logic between OIDC endpoints and these profile endpoints.

Reply:
LOG-10169, it requires lots of work, so I created this issue to track, will implment this before releasing.